### PR TITLE
add GyreJS v0.2.5 with npm auto-update

### LINKED ISSUE #6383: [Request] Add GyreJS
**Library name: GyreJS** 
**Git repository url: https://github.com/WRidder/GyreJS**
**npm package url(optional): https://www.npmjs.com/package/gyrejs** 
**License(s): MIT**
**Official homepage: http://gyre.js.org/**

### PR DESCRIPTION
**Pull request by kennynaoh**
PR for #6383
@dannyxx001  please help me review this pull request, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/WRidder/GyreJS
 * Official website (optional, not the repository): http://gyre.js.org/
 * NPM package url (optional): https://www.npmjs.com/package/gyrejs
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 4
   - Count of watchers: 2
   - Count of forks: 0
 * NPM download stats (optional):
   - Downloads in the last day: 0
   - Downloads in the last week: 404
   - Downloads in the last month: 658

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #6383, cc @WRidder

### CODE REVIEW

File: ajax/libs/gyrejs/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  "npmName": "gyrejs",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/*"
+      ]
+    }
+  ],
+  "tonicExample": "var GyreJS = require('gyrejs');"
```
Comment (PeterDaveHello) [inline ajax/libs/gyrejs/package.json]: Please remove this line.

File: ajax/libs/gyrejs/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  "keywords": [
+    "flux",
+    "reactjs",
+    "immutable",
+    "state",
+    "predictable"
+  ],
+  "author": {
+    "name": "Wilbert van de Ridder"
+  },
```
Comment (PeterDaveHello) [inline ajax/libs/gyrejs/package.json]: `"author": "Wilbert van de Ridder"`

File: ajax/libs/gyrejs/package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "gyrejs",
+  "filename": "gyrejs.min.js",
+  "version": "0.2.5",
+  "description": "Experimental data fetching and state library geared towards ReactJS",
+  "jsnext:main": "src/index.js",
```
Comment (PeterDaveHello) [inline ajax/libs/gyrejs/package.json]: We don't need this.

**Comment by kennynaoh** [comment]
@PeterDaveHello 
Thanks for helping me review this PR
@dannyxx001 
Please help me review it again
Thanks

**Comment by WRidder** [comment]
Thanks guys! Anything I can do at this moment?

**Comment by PeterDaveHello** [comment]
@WRidder you can help us confirm the files need to be hosted if you want :smile:

**Comment by WRidder** [comment]
Well, at this moment, since it's still under development and looking at my roadmap for the upcoming weeks/months, it does not really have to be hosted at this very moment. Maybe a few versions further into the development makes more sense at this point :) 

But I do really appreciate the effort so far and I hope that a few releases out, the process of actually getting it hosted will be smooth since you guys already had a look at it.

**Comment by PeterDaveHello** [comment]
@WRidder thanks for your prompt reply, since there is request to ask adding this lib, if you don't mind, we'll add this lib ASAP, this can also help promote your lib i think :)

**Comment by WRidder** [comment]
That's most certainly true! It's merely a note for you guys that, if other matters are more pressing, this does not have the highest priority at the moment :)

I'm most thankful for the great response here; keep it up!

**Comment by PeterDaveHello** [comment]
@WRidder Thanks.

**Comment by PeterDaveHello** [comment]
Ping @dannyxx001

File: ajax/libs/gyrejs/package.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  "keywords": [
+    "flux",
+    "reactjs",
+    "immutable",
+    "state",
+    "predictable"
+  ],
+  "author": "Wilbert van de Ridder",
+  "license": "MIT",
+  "homepage": "https://github.com/WRidder/gyrejs#readme",
```
Comment (dannyxx001) [inline ajax/libs/gyrejs/package.json]: If the homepage is the same as its' github, you should remove it.

**Comment by kennynaoh** [comment]
@dannyxx001 
I have updated.
Please help me review it again
Thanks

**Comment by kennynaoh** [comment]
ping @dannyxx001

**Comment by dannyxx001** [comment]
I think this PR is OK. @PeterDaveHello please review again, thanks.

**Comment by kennynaoh** [comment]
@dannyxx001 
I have rebased to the latest master branch.
Please help me review it again.
Thanks.

**Pull request merged**